test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+    printf("Iteration %s: Tables %s.%d and %s.%d %s\n", msg, full_out, i, incr_out, i,
+      ret == 0 ? "identical" : "differ");
+    if (ret != 0)
+        exit(1);
+
+    /*
+     * If they compare successfully, clean up.
+     */
+    if (i != 0) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d %s.%d %s.%d %s.%d", home_full,
+          i, home_incr, i, full_out, i, incr_out, i));
+        testutil_check(system(buf));
+    }
+    return (ret);
+}
+
+/*
+ * Set up all the directories needed for the test. We have a full backup directory for each
+ * iteration and an incremental backup for each iteration. That way we can compare the full and
+ * incremental each time through.
+ */
+static void
+setup_directories(void)
+{
+    int i;
+    char buf[1024];
+
+    for (i = 0; i <= MAX_ITERATIONS; i++) {
+        /*
+         * For incremental backups we need 0-N. The 0 incremental directory will compare with the
+         * original at the end.
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_incr, i, home_incr, i, logpath));
+        testutil_check(system(buf));
+        if (i == 0)
+            continue;
+        /*
+         * For full backups we need 1-N.
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_full, i, home_full, i, logpath));
+        testutil_check(system(buf));
+    }
+}
+
+static void
+add_work(WT_SESSION *session, int iter, int iterj, const char *t_uri1, const char *t_uri2)
+{
+    WT_CURSOR *cursor, *cursor2;
+    int i;
+    char k[64], v[64];
+
+    testutil_check(session->open_cursor(session, t_uri1, NULL, NULL, &cursor));
+
+    /*
+     * Only on even iterations add content to the extra table. This illustrates and shows that
+     * sometimes only some tables will be updated.
+     */
+    cursor2 = NULL;
+    if (iter % 2 == 0) {
+        testutil_check(session->open_cursor(session, t_uri2, NULL, NULL, &cursor2));
+    }
+    /*
+     * Perform some operations with individual auto-commit transactions.
+     */
+    for (i = 0; i < MAX_KEYS; i++) {
+        testutil_check(__wt_snprintf(k, sizeof(k), "key.%d.%d.%d", iter, iterj, i));
+        testutil_check(__wt_snprintf(v, sizeof(v), "value.%d.%d.%d", iter, iterj, i));
+        cursor->set_key(cursor, k);
+        cursor->set_value(cursor, v);
+        testutil_check(cursor->insert(cursor));
+        if (cursor2 != NULL) {
+            cursor2->set_key(cursor2, k);
+            cursor2->set_value(cursor2, v);
+            testutil_check(cursor2->insert(cursor2));
+        }
+    }
+    testutil_check(cursor->close(cursor));
+    if (cursor2 != NULL)
+        testutil_check(cursor2->close(cursor2));
+}
+
+static void
+add_bulk_load(WT_SESSION *session, int iter_i, int iter_j, const char *t_uri)
+{
+    WT_CURSOR *cursor;
+    int i;
+    int k;
+    char v[64];
```

Comment:
This is repeat code from `add_work` mostly. Can this be refactored so that this function and `add_work` both call some other function like `add_data` and take cursors as an arg?

Reply:
I have updated the `add_work` function to take an extra boolean argument "bulk_load" to drive the bulk load option.